### PR TITLE
Add tests for deletable_chip_attributes.on_deleted.0.dart

### PR DESCRIPTION
**Pull request by ValentinVignal**
Contributes to https://github.com/flutter/flutter/issues/130459

It adds a test for
- `examples/api/lib/material/chip/deletable_chip_attributes.on_deleted.0.dart`

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: examples/api/test/material/chip/deletable_chip_attributes.on_deleted.0_test.dart [hunk 1]
```diff
@@ -0,0 +1,65 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/chip/deletable_chip_attributes.on_deleted.0.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('RawChip.deleteIconBoxConstraints updates delete icon size constraints', (WidgetTester tester) async {
```
Comment (bleroux) [inline examples/api/test/material/chip/deletable_chip_attributes.on_deleted.0_test.dart:10]: This test description is from the other DeletableChipAttributes examples. Can you update it?

Comment (ValentinVignal) [reply examples/api/test/material/chip/deletable_chip_attributes.on_deleted.0_test.dart]: Arf sorry about that :(

I changed it in [test: Update description](https://github.com/flutter/flutter/pull/152361/commits/319c904c18646c625e971d652e6b8e00bce83b54)

**Review by bleroux — APPROVED** [review]
LGTM!

@ValentinVignal You made a lot of valuable contributions during the past years, would you be interested in getting contributor access (see https://github.com/flutter/flutter/blob/master/docs/contributing/Contributor-access.md)?
If yes, @TahaTesser and I will be happy to sponsor you.

**Comment by ValentinVignal** [comment]
I would love that @bleroux :) That would be awesome!

File: examples/api/test/material/chip/deletable_chip_attributes.on_deleted.0_test.dart [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    expect(find.widgetWithText(AppBar, 'DeletableChipAttributes.onDeleted Sample'), findsOne);
+    expect(find.widgetWithText(Chip, 'Aaron Burr'), findsOne);
+    expect(find.widgetWithText(Chip, 'Alexander Hamilton'), findsOne);
+    expect(find.widgetWithText(Chip, 'Eliza Hamilton'), findsOne);
+    expect(find.widgetWithText(Chip, 'James Madison'), findsOne);
+
+    // Delete Alexander Hamilton.
+    await tester.tap(
+      find.descendant(of: find.widgetWithText(Chip, 'Alexander Hamilton'), matching: find.byIcon(Icons.cancel)),
```
Comment (TahaTesser) [inline examples/api/test/material/chip/deletable_chip_attributes.on_deleted.0_test.dart:23]:
```suggestion
      find.descendant(
        of: find.widgetWithText(Chip, 'Alexander Hamilton'),
        matching: find.byIcon(Icons.cancel),
      ),
```

Comment (TahaTesser) [reply examples/api/test/material/chip/deletable_chip_attributes.on_deleted.0_test.dart:23]: Can we also put in a method that returns `Finder` and reuse in this test.

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: examples/api/test/material/chip/deletable_chip_attributes.on_deleted.0_test.dart (hunk 2 above)
Comment (ValentinVignal) [reply examples/api/test/material/chip/deletable_chip_attributes.on_deleted.0_test.dart]: Sure, is [refactor: Create utility method to find chip delete button](https://github.com/flutter/flutter/pull/152361/commits/fecfa6fbe95092afb65b8f9b2bbdf5275b030eea) fine with you?

File: examples/api/test/material/chip/deletable_chip_attributes.on_deleted.0_test.dart [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    return find.descendant(
+      of: find.widgetWithText(Chip, chipText),
+      matching: find.byIcon(Icons.cancel),
+    );
+  }
+
+  testWidgets('Chip.onDeleted can be used to delete chips', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.OnDeletedExampleApp(),
+    );
```
Comment (TahaTesser) [inline examples/api/test/material/chip/deletable_chip_attributes.on_deleted.0_test.dart:20]:
```suggestion
  testWidgets('Chip.onDeleted can be used to delete chips', (WidgetTester tester) async {
    await tester.pumpWidget(
      const example.OnDeletedExampleApp(),
    );
    
  Finder cancelIconFinder(String chipText) {
    return find.descendant(
      of: find.widgetWithText(Chip, chipText),
      matching: find.byIcon(Icons.cancel),
    );
  }
    
```


Unless the code is reused in multiple tests, it's better to keep inside the test code for readability and portability.

**Review by TahaTesser — APPROVED** [review]
LGTM with nit

**Pull request merged**